### PR TITLE
Update mail-listener2 with changes from another forks

### PR DESCRIPTION
**Pull request by dapanas**
This pull request includes changes and improvements from another forks.

The changes are:
* Avoid to process the same email multiple times
* Fix unread param on empty response
* Add a restart function to reconnect to imap server
* Enhanced logging for the new features
* Pass email object to _attachment_ event
* Update mime dependency version to ^1.0.0 because mime 2.0.0 package break the API

Every commit was made by me giving the right credit to the original authors.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -138,11 +163,23 @@ function parseUnread() {
         f.once('error', function(err) {
           self.emit('error', err);
         });
-      }, function(err){
-        if( err ) {
+      }, function (err) {
+        console.log('all process');
```
Comment (uapasha) [inline index.js:167]: console.log is redundant

File: package.json [hunk 2]
```diff
@@ -1,17 +1,18 @@
 {
-  "name": "mail-listener2",
-  "version": "0.3.1",
-  "description": "Mail listener library for node.js. Get notification when new email arrived.",
+  "name": "mail-listener2-updated",
```
Comment (uapasha) [inline package.json:2]: name is wrong

File: package.json [hunk 3]
```diff
@@ -21,9 +22,9 @@
     "email parser"
   ],
   "author": {
-    "name": "Chirag Jain",
-    "email": "jain_chirag04@yahoo.com",
-    "url": "http://chiragjain.tumblr.com"
+    "name": "Daniel Pérez",
+    "email": "dapanas@gmail.com",
+    "url": "http://blogcito.info"
```
Comment (uapasha) [inline package.json:27]: url, homepage, author are wrong

File: readme.md [hunk 4]
```diff
@@ -1,6 +1,17 @@
...
+Updated Mail-listener2 includes changes and improvements from another forks.
+
+The changes are:
+
+* Avoid to process the same email multiple times
+* Fix unread param on empty response
+* Add a restart function to reconnect to imap server
+* Enhanced logging for the new features
+* Pass email object to attachment event
+* Update mime dependency version to ^1.0.0 because mime 2.0.0 package break the API
```
Comment (uapasha) [inline readme.md:14]: description should be fixed

File: readme.md [hunk 5]
```diff
@@ -10,15 +21,15 @@ Heavily inspired by [mail-listener](https://github.com/circuithub/mail-listener)
...
+`npm install mail-listener2-updated`
 
 
 JavaScript Code:
 
 
 ```javascript
 
-var MailListener = require("mail-listener2");
+var MailListener = require("mail-listener2-updated");
```
Comment (uapasha) [inline readme.md:32]: package name is wrong

File: index.js [hunk 6]
```diff
@@ -52,6 +54,15 @@ MailListener.prototype.stop = function() {
   this.imap.end();
 };
 
+MailListener.prototype.restart = function() {
+  console.log('detaching existing listener');
+  this.imap.removeAllListeners('mail');
+  this.imap.removeAllListeners('update');
+
+  console.log('calling imap connect');
```
Comment (uapasha) [inline index.js:62]: these two consoles probably can be removed

**Review by uapasha — COMMENTED** [review]
Thanks a lot for your work! I hope it would be merged soon

**Comment by dapanas** [comment]
Hi @uapasha,

The latest commit was made after the pull request in order to create a new package for npm. That's why you found another package name, author and url. My bad creating the pull request from master and not from specific commit. Sorry.

The pull request was done with this commit: f6a9fe7.

**Comment by dapanas** [comment]
Closed and replaced by #74.

**Pull request closed without merging**